### PR TITLE
feat(Component/List): actions in a menu dropdown

### PR DESCRIPTION
**Pull request by jsomsanith**
**What is the problem this PR is trying to solve?**
There are a lot of actions in apps list items, and it's still growing.
The UX direction is to put all actions in a [complex tooltip](http://guidelines.talend.com/document/92132#/ui-controls/tooltips) 

**What is the chosen solution to this problem?**
This is a first step towards UX target without breaking changes.
We don't have the complex tooltip, to for now we split simple actions from dropdown actions.

* Put all simple actions in an ellipsis `...` menu.
* Put the dropdown actions next the the ellipsis menu

menu

Please don't complain about the guideline, it's a first step toward the guideline :D

**Please check if the PR fulfills these requirements**

* [ ] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
* [x] Tests for the changes have been added (for bug fixes / features) And [non reg](./screenshots.md) done before need review
* [ ] Docs have been added / updated (for bug fixes / features)
* [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

File: output/cmf-webpack-plugin.eslint.txt [hunk 1]
```diff
@@ -1,3 +1,9 @@
 > eslint --config ../../.eslintrc --ext .js src
 
 The react/require-extension rule is deprecated. Please use the import/extensions rule from eslint-plugin-import instead.
+
+/home/travis/build/Talend/ui/packages/cmf-webpack-plugin/src/index.js
+  4:13  error  'lodash' should be listed in the project's dependencies, not devDependencies  import/no-extraneous-dependencies
```
Comment (jare-talend) [inline output/cmf-webpack-plugin.eslint.txt]: fix lint error

File: packages/components/src/VirtualizedList/CellTitle/CellTitle.scss [hunk 2]
```diff
@@ -47,38 +47,75 @@ $tc-list-title-icon-size: $svg-lg-size !default;
...
-:global(.tc-list-large-row):hover :global(.tc-list-title .tc-actions),
-:global(.ReactVirtualized__Table__row):hover :global(.tc-list-title .tc-actions) {
-	opacity: 1;
-	transition: opacity 0.15s ease-in;
-	visibility: visible;
+:global(.tc-list-large-row),
+:global(.ReactVirtualized__Table__row) {
+	&:hover,
+	&:focus,
+	&:focus-within {
```
Comment (jare-talend) [inline packages/components/src/VirtualizedList/CellTitle/CellTitle.scss]: this feature doesn't work on IE/Edge, do you have a workaround to handle that on this browser ?

File: packages/components/src/VirtualizedList/CellTitle/CellTitleActions.component.js [hunk 3]
```diff
@@ -31,16 +63,21 @@ function CellTitleActions({ rowData, actionsKey, displayMode, persistentActionsK
...
-	rowData: PropTypes.object, // eslint-disable-line
+	rowData: PropTypes.object,
+	t: PropTypes.func.isRequired,
+};
+CellTitleActionsComponent.defaultProps = {
+	t: getDefaultT(),
 };
 
-export default CellTitleActions;
+export default translate(I18N_DOMAIN_COMPONENTS, { i18n: DEFAULT_I18N })(CellTitleActionsComponent);
```
Comment (jare-talend) [inline packages/components/src/VirtualizedList/CellTitle/CellTitleActions.component.js]: Now, define a i18n instance to the HOC translateis not necessary. Just import import '../../translate' is useful to set a default i18n instance to react-i18next.
you can do translate(I18N_DOMAIN_COMPONENTS)(CellTitleActionsComponent);

File: packages/components/src/Actions/ActionDropdown/ActionDropdown.component.js [hunk 4]
```diff
@@ -59,9 +59,15 @@ function renderMutableMenuItem(item, index, getComponent) {
 		return <Renderers.MenuItem key={index} divider />;
 	}
 	return (
-		<Renderers.MenuItem key={index} eventKey={item} {...item} onClick={wrapOnClick(item)}>
+		<Renderers.MenuItem
+			{...item}
+			key={index}
+			eventKey={item}
+			onClick={wrapOnClick(item)}
+			title={item.label}
```
Comment (frassinier) [inline packages/components/src/Actions/ActionDropdown/ActionDropdown.component.js]: the title should not be the label I guess, it should provide more information

File: packages/components/src/VirtualizedList/CellTitle/CellTitleActions.scss [hunk 5]
```diff
@@ -1,33 +1,66 @@
...
+		margin-top: 5px;
+		overflow: visible;
+		left: 50%;
+		transform: translateX(-50%);
+
+		&:before {
+			content: '\25b2';
+			position: absolute;
+			top: -20px;
+			font-size: 1.4em;
```
Comment (frassinier) [inline packages/components/src/VirtualizedList/CellTitle/CellTitleActions.scss]: Stop mixing all units please

File: packages/components/src/VirtualizedList/CellTitle/CellTitle.scss (hunk 2 above)
Comment (frassinier) [reply packages/components/src/VirtualizedList/CellTitle/CellTitle.scss]: It's not worth! :)

File: packages/components/src/VirtualizedList/CellTitle/CellTitleActions.component.js (hunk 3 above)
Comment (jsomsanith) [reply packages/components/src/VirtualizedList/CellTitle/CellTitleActions.component.js]: 👍 this PR was months before your change, guess I have to adapt it :D

**Comment by jsomsanith** [comment]
Let's deal with a11y fix in another PR, I removed the focus-within that made the menu non visible is IE/Edge.

**Comment by jare-talend** [comment]
There is still some issue on IE11
image

**Pull request merged**